filesystem/VFileSystem009.cpp

```diff
@@ -449,10 +445,12 @@ class CXashFS : public IVFileSystem009
 
 	bool AddPackFile( const char *path, const char *id ) override
 	{
-		char *p = va( "%s/%s", IdToDir( id ), path );
-		CopyAndFixSlashes( p, path );
+		string dir, tempbuf;
+
+		Q_snprintf( tempbuf, sizeof(string), "%s/%s", IdToDir( dir, id ), path );
```

Comment:
Let's `sizeof` of variable than it's type.

Variable type may change, but sizeof may be forgotten.

Reply:
This actually applies to every usage of `sizeof` here. It's just too optimistic.